manifests/jenkins.yaml

```diff
@@ -39,21 +39,20 @@ objects:
         storage: ${VOLUME_CAPACITY}
     # DELTA: support specifying storage class
     storageClassName: "${STORAGE_CLASS_NAME}"
-- apiVersion: v1
-  kind: DeploymentConfig
+- apiVersion: apps/v1
+  kind: Deployment
   metadata:
     annotations:
+      image.openshift.io/triggers: '[{"from":{"kind":"ImageStreamTag","name":"${JENKINS_IMAGE_STREAM_TAG}"},"namespace":"${NAMESPACE}","fieldPath":"spec.template.spec.containers[?(@.name==\"${JENKINS_SERVICE_NAME}\")].image","paused":false}]'
       template.alpha.openshift.io/wait-for-ready: "true"
     name: ${JENKINS_SERVICE_NAME}
   spec:
     replicas: 1
     selector:
-      name: ${JENKINS_SERVICE_NAME}
+      matchLabels:
+        name: ${JENKINS_SERVICE_NAME}
     strategy:
       type: Recreate
-      # DELTA: allow 20 mins for pod to be ready to account for slow NFS
-      recreateParams:
```

Comment:
A suggestion, you can use a startup probe https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes to maintain this behavior; otherwise, I suppose you'd face a lot of pod restarts and noise.

Reply:
Great catch, honestly I didn't know how to replace the above functionality. Thanks for the suggestion - fixed.